src/clients/compositor/mod.rs

```diff
@@ -146,10 +171,20 @@ pub enum WorkspaceUpdate {
 
 pub trait WorkspaceClient: Debug + Send + Sync {
     /// Requests the workspace with this name is focused.
-    fn focus(&self, name: String) -> Result<()>;
+    fn focus(&self, name: String);
 
     /// Creates a new to workspace event receiver.
     fn subscribe_workspace_change(&self) -> broadcast::Receiver<WorkspaceUpdate>;
 }
 
 register_fallible_client!(dyn WorkspaceClient, workspaces);
+
+pub trait KeyboardLayoutClient: Debug + Send + Sync {
+    /// Switches to the next layout.
+    fn next_keyboard_layout(&self);
+
+    /// Creates a new to keyboard layout event receiver.
+    fn subscribe_keyboard_layout_change(&self) -> broadcast::Receiver<KeyboardLayoutUpdate>;
```

Comment:
I think these method names can be improved as they're a bit long and partially redundant. `subscribe` matches the other client names. `set_next_active` I think is a better descriptor as `next_keyboard_layout` doesn't necessarily convey it's switching (sounds more like a getter than setter)

```suggestion
    /// Switches to the next layout.
    fn set_next_active(&self);

    /// Creates a new to keyboard layout event receiver.
    fn subscribe(&self) -> broadcast::Receiver<KeyboardLayoutUpdate>;
```

Reply:
> `subscribe` matches the other client names. 

`WorkspaceClient` has `subscribe_workspace_change`. I changed it too.

Potentially this may cause problems when using `sway::Client` directly (not through `dyn WorkspaceClient` or `dyn KeyboardLayoutClient`), because it now has two `subscribe` methods from different traits.